package/hunter/HunterInit.cmake

```diff
@@ -0,0 +1,14 @@
+if(NOT EXISTS ${CMAKE_CURRENT_LIST_DIR}/HunterGate.cmake)
+  file(
+    DOWNLOAD
+    https://raw.githubusercontent.com/hunter-packages/gate/c4e5b3e4d1c97fb46ae86af7f59baac95f7f1cf0/cmake/HunterGate.cmake
+    ${CMAKE_CURRENT_LIST_DIR}//HunterGate.cmake
```

Comment:
See https://github.com/mosra/magnum/pull/298/commits/1af4ef026521f763578bbd9bf1970244b10a8d85 : it checks the status, the sha256 and prints nice messages on failure.

Reply:
> See 1af4ef0 : it checks the status, the sha256 and prints nice messages on failure

My real point here was that adding HunterGate.cmake to sources is the simpiest and robust solution. For example your current code do not support download retry logic and if somebody will interrupt download procedure with Ctrl+C you will have broken `HunterGate.cmake` and `if(NOT EXISTS ${hunter_gate_file})` logic will not allow to calculate hash of it.

Also note that your approach require to always have connection to network, it's not a requirement for Hunter. Both Hunter archives and packages sources are ok to be on local machine only.

Reply:
@ruslo yes, I know. Don't want to sound mean here, my request here was simply to avoid having the file in the project. With Magnum I go to extreme lengths to reduce non-essential features and boilerplate and that goes also for the buildsystem.

For me, cases like abort in the middle of a download or broken checksums fall into the broad category of rare "CMake build dir is totally hosed" problems and the usual solution is to wipe the build directory and start over. When Ithe network connection requirement or broken HunterGate files become a real problem for users, I'll revisit this; until then I'm not sure this is worth spending too much time on.